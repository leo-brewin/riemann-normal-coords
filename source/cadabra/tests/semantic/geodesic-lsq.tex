\def\Date{19 Jan 2024}
% \def\FileID{file:}

\documentclass[12pt]{cdblatex}

\begin{document}

\section*{\jobname}

\CdbSetup{action=hide}

\begin{cadabra}
   import shared

   import cdblib

   term00A = cdblib.get ('check000','expected/geodesic-lsq.json')
   term01A = cdblib.get ('check001','expected/geodesic-lsq.json')
   term02A = cdblib.get ('check002','expected/geodesic-lsq.json')
   term03A = cdblib.get ('check003','expected/geodesic-lsq.json')
   term04A = cdblib.get ('check004','expected/geodesic-lsq.json')
   term05A = cdblib.get ('check005','expected/geodesic-lsq.json')

   term00B = cdblib.get ('check000','output/geodesic-lsq.json')
   term01B = cdblib.get ('check001','output/geodesic-lsq.json')
   term02B = cdblib.get ('check002','output/geodesic-lsq.json')
   term03B = cdblib.get ('check003','output/geodesic-lsq.json')
   term04B = cdblib.get ('check004','output/geodesic-lsq.json')
   term05B = cdblib.get ('check005','output/geodesic-lsq.json')

   diff000 = shared.check (term00A,term00B)   # cdb (diff000,diff000)
   diff001 = shared.check (term01A,term01B)   # cdb (diff001,diff001)
   diff002 = shared.check (term02A,term02B)   # cdb (diff002,diff002)
   diff003 = shared.check (term03A,term03B)   # cdb (diff003,diff003)
   diff004 = shared.check (term04A,term04B)   # cdb (diff004,diff004)
   diff005 = shared.check (term05A,term05B)   # cdb (diff005,diff005)

\end{cadabra}

\begin{dgroup*}
   \Dmath*{ \cdb*{diff000} }
   \Dmath*{ \cdb*{diff001} }
   \Dmath*{ \cdb*{diff002} }
   \Dmath*{ \cdb*{diff003} }
   \Dmath*{ \cdb*{diff004} }
   \Dmath*{ \cdb*{diff005} }
\end{dgroup*}

\end{document}
