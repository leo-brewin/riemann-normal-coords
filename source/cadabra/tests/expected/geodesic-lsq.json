{
  "lsq2": "Dx^{a} Dx^{b} g_{a b} -  1/3 x^{a} x^{b} Dx^{c} Dx^{d} R_{a c b d}",
  "lsq3": "Dx^{a} Dx^{b} g_{a b} -  1/3 x^{a} x^{b} Dx^{c} Dx^{d} R_{a c b d} -  1/12 x^{a} x^{b} Dx^{c} Dx^{d} Dx^{e} \\nabla_{c}(R_{a d b e}) -  1/6 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} \\nabla_{a}(R_{b d c e})",
  "lsq4": "Dx^{a} Dx^{b} g_{a b} -  1/3 x^{a} x^{b} Dx^{c} Dx^{d} R_{a c b d} -  1/12 x^{a} x^{b} Dx^{c} Dx^{d} Dx^{e} \\nabla_{c}(R_{a d b e}) -  1/6 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} \\nabla_{a}(R_{b d c e}) -  1/45 x^{a} x^{b} Dx^{c} Dx^{d} Dx^{e} Dx^{f} g^{g h} R_{a c d g} R_{b e f h} +  2/45 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} g^{g h} R_{a d b g} R_{c e f h} -  1/40 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} \\nabla_{d a}(R_{b e c f}) -  1/40 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} \\nabla_{a d}(R_{b e c f}) -  1/60 x^{a} x^{b} Dx^{c} Dx^{d} Dx^{e} Dx^{f} \\nabla_{c d}(R_{a e b f}) +  2/45 x^{a} x^{b} x^{c} x^{d} Dx^{e} Dx^{f} g^{g h} R_{a e b g} R_{c f d h} -  1/20 x^{a} x^{b} x^{c} x^{d} Dx^{e} Dx^{f} \\nabla_{a b}(R_{c e d f})",
  "lsq5": "Dx^{a} Dx^{b} g_{a b} -  1/3 x^{a} x^{b} Dx^{c} Dx^{d} R_{a c b d} -  1/12 x^{a} x^{b} Dx^{c} Dx^{d} Dx^{e} \\nabla_{c}(R_{a d b e}) -  1/6 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} \\nabla_{a}(R_{b d c e}) -  1/45 x^{a} x^{b} Dx^{c} Dx^{d} Dx^{e} Dx^{f} g^{g h} R_{a c d g} R_{b e f h} +  2/45 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} g^{g h} R_{a d b g} R_{c e f h} -  1/40 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} \\nabla_{d a}(R_{b e c f}) -  1/40 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} \\nabla_{a d}(R_{b e c f}) -  1/60 x^{a} x^{b} Dx^{c} Dx^{d} Dx^{e} Dx^{f} \\nabla_{c d}(R_{a e b f}) +  2/45 x^{a} x^{b} x^{c} x^{d} Dx^{e} Dx^{f} g^{g h} R_{a e b g} R_{c f d h} -  1/20 x^{a} x^{b} x^{c} x^{d} Dx^{e} Dx^{f} \\nabla_{a b}(R_{c e d f}) -  1/270 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} Dx^{g} g^{h i} R_{a d e h} \\nabla_{f}(R_{b g c i}) -  1/45 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} Dx^{g} g^{h i} R_{a d e h} \\nabla_{b}(R_{c f g i}) +  1/108 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} Dx^{g} g^{h i} R_{a d e h} \\nabla_{i}(R_{b f c g}) -  1/60 x^{a} x^{b} Dx^{c} Dx^{d} Dx^{e} Dx^{f} Dx^{g} g^{h i} R_{a c d h} \\nabla_{e}(R_{b f g i}) +  1/45 x^{a} x^{b} x^{c} x^{d} Dx^{e} Dx^{f} Dx^{g} g^{h i} R_{a e f h} \\nabla_{b}(R_{c g d i}) +  1/45 x^{a} x^{b} x^{c} x^{d} Dx^{e} Dx^{f} Dx^{g} g^{h i} R_{a e b h} \\nabla_{f}(R_{c g d i}) +  1/45 x^{a} x^{b} x^{c} x^{d} Dx^{e} Dx^{f} Dx^{g} g^{h i} R_{a e b h} \\nabla_{c}(R_{d f g i}) -  1/180 x^{a} x^{b} x^{c} x^{d} Dx^{e} Dx^{f} Dx^{g} \\nabla_{e a b}(R_{c f d g}) -  1/180 x^{a} x^{b} x^{c} x^{d} Dx^{e} Dx^{f} Dx^{g} \\nabla_{a e b}(R_{c f d g}) -  1/180 x^{a} x^{b} x^{c} x^{d} Dx^{e} Dx^{f} Dx^{g} \\nabla_{a b e}(R_{c f d g}) +  2/135 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} Dx^{g} g^{h i} R_{a d b h} \\nabla_{e}(R_{c f g i}) -  1/270 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} Dx^{g} \\nabla_{d e a}(R_{b f c g}) -  1/270 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} Dx^{g} \\nabla_{d a e}(R_{b f c g}) -  1/270 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} Dx^{g} \\nabla_{a d e}(R_{b f c g}) -  1/360 x^{a} x^{b} Dx^{c} Dx^{d} Dx^{e} Dx^{f} Dx^{g} \\nabla_{c d e}(R_{a f b g}) +  2/45 x^{a} x^{b} x^{c} x^{d} x^{e} Dx^{f} Dx^{g} g^{h i} R_{a f b h} \\nabla_{c}(R_{d g e i}) -  1/90 x^{a} x^{b} x^{c} x^{d} x^{e} Dx^{f} Dx^{g} \\nabla_{a b c}(R_{d f e g})",
  "lsq50": "Dx^{a} Dx^{b} g_{a b}",
  "lsq52": " -  1/3 x^{a} x^{b} Dx^{c} Dx^{d} R_{a c b d}",
  "lsq53": " -  1/12 x^{a} x^{b} Dx^{c} Dx^{d} Dx^{e} \\nabla_{c}(R_{a d b e}) -  1/6 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} \\nabla_{a}(R_{b d c e})",
  "lsq54": " -  1/45 x^{a} x^{b} Dx^{c} Dx^{d} Dx^{e} Dx^{f} g^{g h} R_{a c d g} R_{b e f h} +  2/45 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} g^{g h} R_{a d b g} R_{c e f h} -  1/40 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} \\nabla_{d a}(R_{b e c f}) -  1/40 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} \\nabla_{a d}(R_{b e c f}) -  1/60 x^{a} x^{b} Dx^{c} Dx^{d} Dx^{e} Dx^{f} \\nabla_{c d}(R_{a e b f}) +  2/45 x^{a} x^{b} x^{c} x^{d} Dx^{e} Dx^{f} g^{g h} R_{a e b g} R_{c f d h} -  1/20 x^{a} x^{b} x^{c} x^{d} Dx^{e} Dx^{f} \\nabla_{a b}(R_{c e d f})",
  "lsq55": " -  1/270 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} Dx^{g} g^{h i} R_{a d e h} \\nabla_{f}(R_{b g c i}) -  1/45 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} Dx^{g} g^{h i} R_{a d e h} \\nabla_{b}(R_{c f g i}) +  1/108 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} Dx^{g} g^{h i} R_{a d e h} \\nabla_{i}(R_{b f c g}) -  1/60 x^{a} x^{b} Dx^{c} Dx^{d} Dx^{e} Dx^{f} Dx^{g} g^{h i} R_{a c d h} \\nabla_{e}(R_{b f g i}) +  1/45 x^{a} x^{b} x^{c} x^{d} Dx^{e} Dx^{f} Dx^{g} g^{h i} R_{a e f h} \\nabla_{b}(R_{c g d i}) +  1/45 x^{a} x^{b} x^{c} x^{d} Dx^{e} Dx^{f} Dx^{g} g^{h i} R_{a e b h} \\nabla_{f}(R_{c g d i}) +  1/45 x^{a} x^{b} x^{c} x^{d} Dx^{e} Dx^{f} Dx^{g} g^{h i} R_{a e b h} \\nabla_{c}(R_{d f g i}) -  1/180 x^{a} x^{b} x^{c} x^{d} Dx^{e} Dx^{f} Dx^{g} \\nabla_{e a b}(R_{c f d g}) -  1/180 x^{a} x^{b} x^{c} x^{d} Dx^{e} Dx^{f} Dx^{g} \\nabla_{a e b}(R_{c f d g}) -  1/180 x^{a} x^{b} x^{c} x^{d} Dx^{e} Dx^{f} Dx^{g} \\nabla_{a b e}(R_{c f d g}) +  2/135 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} Dx^{g} g^{h i} R_{a d b h} \\nabla_{e}(R_{c f g i}) -  1/270 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} Dx^{g} \\nabla_{d e a}(R_{b f c g}) -  1/270 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} Dx^{g} \\nabla_{d a e}(R_{b f c g}) -  1/270 x^{a} x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} Dx^{g} \\nabla_{a d e}(R_{b f c g}) -  1/360 x^{a} x^{b} Dx^{c} Dx^{d} Dx^{e} Dx^{f} Dx^{g} \\nabla_{c d e}(R_{a f b g}) +  2/45 x^{a} x^{b} x^{c} x^{d} x^{e} Dx^{f} Dx^{g} g^{h i} R_{a f b h} \\nabla_{c}(R_{d g e i}) -  1/90 x^{a} x^{b} x^{c} x^{d} x^{e} Dx^{f} Dx^{g} \\nabla_{a b c}(R_{d f e g})"
}
